domains/transfer/TransferAgreementAutomation/Worker.UnitTests/Service/ProjectOriginWalletServiceTest.cs

```diff
@@ -138,18 +153,23 @@ public async Task
         TransferCertificates_CertificateEndDateAfterTAStartDatCertificateStartDateIsBefore_ShouldNotCallWalletTransferCertificate()
     {
         var now = DateTimeOffset.UtcNow;
-        var transferAgreement = new TransferAgreementDto(
-            EndDate: DateTimeOffset.UtcNow.AddHours(3).ToUnixTimeSeconds(),
-            ReceiverReference: Guid.NewGuid().ToString(),
-            ReceiverTin: "12345678",
-            SenderId: Guid.NewGuid().ToString(),
-            StartDate: now.ToUnixTimeSeconds()
-        );
+        var transferAgreement = new TransferAgreement
+        {
+            EndDate = DateTimeOffset.UtcNow.AddHours(3),
+            ReceiverReference = Guid.NewGuid(),
+            ReceiverTin = "12345678",
+            SenderId = Guid.NewGuid(),
+            StartDate = now,
+            Id = Guid.NewGuid(),
+            SenderName = "SomeSender",
+            SenderTin = "11223344",
+            TransferAgreementNumber = 0
+        };
 
         var fakeGranularCertificatesResponse = CreateAsyncUnaryCall(
-            new QueryResponse
-            { GranularCertificates = { CreateGranularCertificate(now.AddHours(-1), now.AddHours(1)) } }
-        );
+                new QueryResponse
```

Comment:
Why does the concept of certificates exist in the transfer subsystem? Are we crossing a service boundary here?

Reply:
Because it transfers certificates?